### ISSUE TITLE
Sanitize the "Notes" field

### ISSUE THREAD

**Issue by rgreinho** [scrapd/scrapd#63]
Issue Type
----------

- Feature request

Current Behavior
----------------

The extraction of the information from the "Notes" field works fairly well, but the structure of the data is so random it does not always provide a very readable output.

For example:
```json
{
"Age": 15,
"Case": "19-0081623",
"DOB": "02/18/2003",
"Date": "01/08/2019",
"Ethnicity": "Hispanic",
"Fatal crashes this year": "3",
"First Name": "Jesus",
"Gender": "male",
"Last Name": "Servantez",
"Link": "http://austintexas.gov/news/traffic-fatality-3-4",
"Location": "3600 block of South Capital of Texas Highway SB",
"Notes": "Jesus Servantez, Hispanic male, DOB 02/18/2003<br /> \t <br /> \tThe preliminary investigation shows that the driver of a silver, 2018 KIA was traveling in the center lane of the 3600 block of South Capital of Texas Highway SB when the car collided with Jesus Servantez, a pedestrian in the roadway. The driver remained at the scene and told investigators that he did not see Servantez prior to the crash.<br /> \t <br /> \tJesus Servantez was transported to Saint David\u2019s South Austin Medical Center where he succumbed to his injuries on January 21, 2019. No charges are expected to be filed.<br /> \t <br /> \tAPD is investigating this case. Anyone with information regarding this case should call APD\u2019s Vehicular Homicide Unit Detectives at (512) 974-3761. You can also submit tips by downloading APD\u2019s mobile app, Austin PD, for free on <a href=\"https://austintexas.us5.list-manage.com/track/click?u=1861810ce1dca1a4c1673747c&amp;id=fefeacb035&amp;e=bcdeacc118\">iPhone</a> and <a href=\"https://austintexas.us5.list-manage.com/track/click?u=1861810ce1dca1a4c1673747c&amp;id=96aaeb3f00&amp;e=bcdeacc118\">Android</a>.<br /> \t <br /> \tThis is Austin\u2019s third fatal traffic crash of 2018, resulting in three fatalities this year. At this time in 2018, there were two fatal traffic crashes and three traffic fatalities.<br /> \t <br /><strong><em>These statements are based on the initial assessment of the fatal crash and investigation is still pending. Fatality information may change.</em></strong></p> </div></div></div><div class=\"service-links\"><div class=\"service-label\">Share </div><ul class=\"links\"><li class=\"service-links-facebook first\"><a href=\"http://www.facebook.com/sharer.php?u=http%3A//austintexas.gov/news/traffic-fatality-3-4&amp;t=Traffic%20Fatality%20%233\" title=\"Share on Facebook.\" class=\"service-links-facebook\" rel=\"nofollow\" target=\"_blank\"><img typeof=\"foaf:Image\" src=\"http://austintexas.gov/sites/all/modules/service_links/images/facebook.png\" alt=\"Facebook\" /></a></li> <li class=\"service-links-twitter\"><a href=\"http://twitter.com/share?url=http%3A//austintexas.gov/news/traffic-fatality-3-4&amp;text=Traffic%20Fatality%20%233\" title=\"Share this on Twitter\" class=\"service-links-twitter\" rel=\"nofollow\" target=\"_blank\"><img typeof=\"foaf:Image\" src=\"http://austintexas.gov/sites/all/modules/service_links/images/twitter.png\" alt=\"Twitter\" /></a></li> <li class=\"service-links-print last\"><a href=\"http://austintexas.gov/print/news/traffic-fatality-3-4\" title=\"Printable version\" class=\"service-links-print\" rel=\"nofollow\" target=\"_blank\"><img typeof=\"foaf:Image\" src=\"http://austintexas.gov/sites/all/modules/service_links/images/print.png\" alt=\"Print HTML\" /></a></li> </ul></div> </div> </div> </div> </div> </div> </div> </div> <div class=\"panel-separator\"></div><div class=\"panel-pane pane-custom pane-1\" > <div class=\"pane-content\"> Communications and Public Information Office 301 W. 2nd Street, Austin, TX 78701 </div> </div> \t </div> \t \t \t \t \t <div class=\"clear\"></div> \t \t</div><!-- / 12 --> </div> </div> </div> </div> \t\t </div> \t\t \t\t \t\t \t </div> \t \t</div><!-- /#maincontent --> </div> <!-- /#page .container-16 --> <div id=\"footer\"> \t\t\t\t\t\t<div id=\"footer-blocks\" class=\"container-16 clearfix\"> <div class=\"region region-footer\"> <div id=\"block-block-28\" class=\"block block-block \"> <div class=\"content\"> <script> </script> </div> </div> <div id=\"block-block-2\" class=\"block block-block grid-2\"> <div class=\"content\"> <p><img alt=\"Official seal of the City of Austin\" src=\"/sites/all/themes/austintexas_ninesixty/images/austin_stamp.png\" />"
}
```

* The first line contains the victim's details that were already extracted in other fields
* The next part is the actual note information. Except for a few `\t` and HTML tags it looks pretty good.
* The final part, after "Fatality information may change" is the footer of the page and should not be there

Expected Behavior
-----------------

Ideally we should be able to extract only the text from the notes. 

Action items:
* Find a way to consistently remove the footer information
* Clean the Notes from all HTML tags and `\t`
* Remove the victims First, Last, Ethnicity, Gender details (i.e. the first line in the example)

**Comment by rgreinho** [comment]
This simple one liner:
```python
''.join(d.get('Notes').split('<br /> \t <br />')[1:-1]).replace('\t','').strip()
```

Actually produces a decent result:
> 'The preliminary investigation shows that the driver of a silver, 2018 KIA was traveling in the center lane of the 3600 block of South Capital of Texas Highway
SB when the car collided with Jesus Servantez, a pedestrian in the roadway. The driver remained at the scene and told investigators that he did not see Servante
z prior to the crash. Jesus Servantez was transported to Saint David’s South Austin Medical Center where he succumbed to his injuries on January 21, 2019. No ch
arges are expected to be filed. APD is investigating this case. Anyone with information regarding this case should call APD’s Vehicular Homicide Unit Detectives
 at (512) 974-3761. You can also submit tips by downloading APD’s mobile app, Austin PD, for free on <a href="https://austintexas.us5.list-manage.com/track/clic
k?u=1861810ce1dca1a4c1673747c&amp;id=fefeacb035&amp;e=bcdeacc118">iPhone</a> and <a href="https://austintexas.us5.list-manage.com/track/click?u=1861810ce1dca1a4
c1673747c&amp;id=96aaeb3f00&amp;e=bcdeacc118">Android</a>. This is Austin’s third fatal traffic crash of 2018, resulting in three fatalities this year. At this
time in 2018, there were two fatal traffic crashes and three traffic fatalities.'

**Comment by rgreinho** [comment]
A regex would be a little bit more flexible:

```python
''.join(re.split('<br />\s*\\t\s*<br />', d.get('Notes'))[1:-1]).replace('\t','').strip()
```

**Comment by rgreinho** [comment]
And another regex to replace the HTML anchor tags:

```python
re.sub('<a[^>]*>([^<]+)</a>', r'\1',snotes)
```

**Comment by rgreinho** [comment]
Assigned to @mrengler.

**Comment by rgreinho** [comment]
Closed by #64.
